extgen: (types) make go arrays more consistent with PHP arrays (#1800)
dunglas/frankenphp: docs/extensions.md

#### Working with Arrays

FrankenPHP provides native support for PHP arrays through `frankenphp.AssociativeArray` and `frankenphp.PackedArray`.

> `AssociativeArray` represents an associative hashmap and is composed of a `Map: map[string]any` and an optional `Order: []string`.

Reviewer: ```suggestion
`AssociativeArray` represents a [hash map](https://en.wikipedia.org/wiki/Hash_table) composed of a `Map: map[string]any`field and an optional `Order: []string` field (unlike PHP "associative arrays", Go maps aren't ordered).
```

For unordered arrays, `Order` will be left empty.

> `PackedArray` is just an alias for a go slice `[]any`

Reviewer: ```suggestion
`PackedArray` is an alias for a Go slice `[]any`
```

		"key2": "value2",
	}})
}

// export_php:function process_data_packed(array $input): array
func process_data_packed(arr *C.zval) unsafe.Pointer {
	// Convert PHP packed array to Go
	packedArray := frankenphp.GoPackedArray(unsafe.Pointer(arr), false)

	// loop over the slice in order
	for index, value := range packedArray {
		// do something with index and value
	}

	// return a packed array
	return frankenphp.PHPackedArray{"value1", "value2", "value3"}
}
```

**Key features of `frankenphp.PackedArray` and `frankenphp.AssociativeArray`:**

* **Ordered key-value pairs** - Option to keep the order of the associative array
* **Optimized for multiple cases** - Option to ditch the order for better performance or convert straight to a slice
* **Automatic list detection** - When converting to PHP, automatically detects if array should be a packed list or hashmap
* **Nested Arrays** - Arrays can be nested and will convert all support types automatically (int64,float64,string,bool,nil,AssociativeArray,PackedArray)
* **Objects are not supported** - Currently, only scalar types and arrays can be used as values. Providing an object will result in a `null` value in the PHP array.

> ##### Available methods - packed and associative

Reviewer: ```suggestion
##### Available methods: Packed and Associative
```
